server/server.go

```diff
@@ -543,143 +432,152 @@ func (s *server) CreateBlockchains(
 		chainSpecs = append(chainSpecs, chainSpec)
 	}
 
-	// check that defined subnets exist
-	subnetsMap := map[string]struct{}{}
-	for _, subnet := range s.clusterInfo.Subnets {
-		subnetsMap[subnet] = struct{}{}
-	}
-	for _, chainSpec := range chainSpecs {
-		if chainSpec.SubnetID != nil {
-			_, ok := subnetsMap[*chainSpec.SubnetID]
-			if !ok {
-				return nil, fmt.Errorf("subnet id %q does not exits", *chainSpec.SubnetID)
-			}
-		}
-	}
+	// check that the given subnets exist
+	subnetsSet := set.Set[string]{}
+	subnetsSet.Add(s.clusterInfo.Subnets...)
 
-	// if there will be a restart, network will not be healthy
-	// until finishing
 	for _, chainSpec := range chainSpecs {
-		if chainSpec.SubnetID == nil {
-			s.clusterInfo.Healthy = false
+		if chainSpec.SubnetID != nil && !subnetsSet.Contains(*chainSpec.SubnetID) {
+			return nil, fmt.Errorf("subnet id %q does not exits", *chainSpec.SubnetID)
 		}
 	}
 
 	s.clusterInfo.Healthy = false
 	s.clusterInfo.CustomChainsHealthy = false
 
-	// start non-blocking to install custom chains (if applicable)
-	// the user is expected to poll cluster status
-	readyCh := make(chan struct{})
-	go s.network.createBlockchains(ctx, chainSpecs, readyCh)
-
 	// update cluster info non-blocking
 	// the user is expected to poll this latest information
 	// to decide cluster/subnet readiness
 	go func() {
-		s.waitChAndUpdateClusterInfo("custom chains", readyCh, true)
+		ctx, cancel := context.WithTimeout(context.Background(), waitForHealthyTimeout)
+		defer cancel()
+		err := s.network.CreateChains(ctx, chainSpecs)
```

Comment:
we want this error to be returned by WaitForHealthy

Reply:
https://github.com/ava-labs/avalanche-network-runner/pull/447#discussion_r1120195954